cime_config/hist_config.py

```diff
@@ -0,0 +1,1126 @@
+
+    def set_output_frequency(self, ofreq, pobj, logger):
+        """Modify the output_frequency property of this HistoryVolConfig
+            object. <ofreq> is a tuple consisting of an integer and a period.
+        """
+        if ( isinstance(ofreq, tuple) and (len(ofreq) == 2) and
+             isinstance(ofreq[0], int) and isinstance(ofreq[1], str) and
+             (ofreq[0] > 0) and
+             (ofreq[1].strip().lower() in _TIME_PERIODS)):
+            self.__output_freq = ofreq
+            if logger.getEffectiveLevel() <= logging.DEBUG:
+                ctx = context_string(pobj)
+                logger.debug(f"Setting output_frequency to '{ofreq}'{ctx}")
+            # end if
+            return True
+        # end if
+        emsg = f"Attempt to set unrecognized output_frequency, '{ofreq}'"
+        pobj.add_syntax_err(emsg)
+        return False
+
+    @property
+    def file_type(self):
+        """Return the file_type property for this HistoryVolConfig object"""
+        return self.__file_type
+
+    def set_file_type(self, ftype, pobj, logger):
+        """Modify the file_type property of this HistoryVolConfig object"""
+        if ftype.strip().lower() in self.__HFILE_TYPES:
+            self.__file_type = ftype
+        else:
+            tstr = f", must be one of ({', '.join(self.__HFILE_TYPES)})."
+            pobj.add_syntax_err(f"Bad history file type, '{ftype}'{tstr}")
+            return False
+        # end if
+        if (ftype == self.__INITIAL_FILE) and (not self.__max_frames_set):
+            self.__max_frames = 1
+        # end if
+        if (ftype == self.__INITIAL_FILE) and (not self.__precision_set):
+            self.__precision = 'REAL64'
+        # end if
+        if (logger is not None) and (logger.getEffectiveLevel() <= logging.DEBUG):
+            ctx = context_string(pobj)
+            logger.debug(f"Setting file type to '{ftype}'{ctx}")
+        # end if
+        return True
+
+    @property
+    def filename_spec(self):
+        """Return the filename_spec property for this HistoryVolConfig object"""
+        return self.__filename_spec
+
+    def set_filename_spec(self, fnspec, pobj, logger):
+        """Modify the filename_spec property of this HistoryVolConfig object.
+        If the restart filename spec has not yet been set, set it to the default
+        for <fnspec> if possible (i.e., if <fnspec> contains a '%u').
+        """
+        self.__filename_spec = fnspec
+        if not self.__restart_fname_spec_set:
+            if '%u' in self.__filename_spec:
+                self.__restart_fname_spec = self.__filename_spec.replace("%u",
+                                                                         "r%u")
+            # end if
+        # end if
+        if logger.getEffectiveLevel() <= logging.DEBUG:
+            ctx = context_string(pobj)
+            logger.debug(f"Setting filename spec to '{fnspec}'{ctx}")
+        # end if
+        return True
```

Comment:
I might be misunderstanding the design but is there a reason to have a return statement if it will always be `True`?  And should we return `False` in certain cases like if `self.__restart_fname_spec_set`?

Reply:
return statement removed.